core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java

```diff
@@ -54,10 +72,84 @@ public KitRequest requestKit(AdminUser adminUser, Enrollee enrollee, String kitT
         return kitRequest;
     }
 
+    /**
+     * Collect the address fields sent to DSM with a kit request. This is not the full DSM request, just the address
+     * information captured at the time of, and stored with, the kit request.
+     */
+    public static PepperKitAddress makePepperKitAddress(Profile profile) {
+        MailingAddress mailingAddress = profile.getMailingAddress();
+        return PepperKitAddress.builder()
+                .firstName(profile.getGivenName())
+                .lastName(profile.getFamilyName())
+                .street1(mailingAddress.getStreet1())
+                .street2(mailingAddress.getStreet2())
+                .city(mailingAddress.getCity())
+                .state(mailingAddress.getState())
+                .postalCode(mailingAddress.getPostalCode())
+                .country(mailingAddress.getCountry())
+                .phoneNumber(profile.getPhoneNumber())
+                .build();
+    }
+
+    /**
+     * Fetch all kits for an enrollee.
+     */
     public Collection<KitRequest> getKitRequests(AdminUser adminUser, Enrollee enrollee) {
         return dao.findByEnrollee(enrollee.getId());
     }
 
+    /**
+     * Query DSM for the status of a single kit and update the cached status in Juniper.
+     * This is intended for the special case of needing the absolute most up-to-date information for a single kit.
+     * Do _NOT_ call this repeatedly for a collection of kits. Use a bulk operation instead to avoid overwhelming DSM.
+     */
+    @Transactional
+    public PepperDSMKitStatus updateKitStatus(UUID kitId) {
+        var kitRequest = dao.find(kitId).orElseThrow(() -> new NotFoundException("Kit request not found"));
+        var pepperDSMKitStatus = pepperDSMClient.fetchKitStatus(kitId);
+        try {
+            kitRequest.setDsmStatus(objectMapper.writeValueAsString(pepperDSMKitStatus));
+            kitRequest.setDsmStatusFetchedAt(Instant.now());
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException("Could not parse JSON response from DSM", e);
+        }
+        dao.update(kitRequest);
+        return pepperDSMKitStatus;
+    }
+
+    /**
+     * Query DSM for all in-progress kits and update the cached status in Juniper. This is intended to be called as a
+     * scheduled job during expected non-busy times for DSM. If on-demand updates are needed outside the scheduled job,
+     * use {@link KitRequestService#updateKitStatus(AdminUser, UUID)} for a single kit or a batch operation that queries
+     * DSM for less than all open kits.
+     */
+    @Transactional
+    public void updateAllKitStatuses() {
+        var studies = studyDao.findAll();
+        for (Study study : studies) {
```

Comment:
Since Pepper (for now at least) will only have one study for all Juniper environments, this is still going by study so that it only queries DSM once per study.